reco_utils/dataset/pandas_df_utils.py

```diff
@@ -117,55 +117,120 @@ def libffm_converter(df, col_rating=DEFAULT_RATING_COL, filepath=None):
         4       1  1:3:1  2:2:7  3:3:5.0  4:8:1
 
     Args:
-        df (pd.DataFrame): input Pandas dataframe.
-        col_rating (str): rating of the data.
         filepath (str): path to save the converted data.
 
-    Return:
-        pd.DataFrame: data in libffm format.
+    Attributes:
+        field_count (int): count of field in the libffm format data
+        feature_count (int): count of feature in the libffm format data
+        filepath (str or None): file path where the output is stored - it can be None or a string
     """
-    df_new = df.copy()
-
-    # Check column types.
-    types = df_new.dtypes
-    if not all([x == object or np.issubdtype(x, np.integer) or x == np.float for x in types]):
-        raise TypeError("Input columns should be only object and/or numeric types.")
-
-    field_names = list(df_new.drop(col_rating, axis=1).columns)
-
-    # Encode field-feature.
-    idx = 1
-    field_feature_dict = {}
-    for field in field_names:
-        if df_new[field].dtype == object:
-            for feature in df_new[field].values:
-                # Check whether (field, feature) tuple exists in the dict or not.
-                # If not, put them into the key-values of the dict and count the index.
-                if (field, feature) not in field_feature_dict:
-                    field_feature_dict[(field, feature)] = idx
-                    idx += 1
-
-    def _convert(field, feature, field_index, field_feature_index_dict):
-        if isinstance(feature, str):
-            field_feature_index = field_feature_index_dict[(field, feature)]
-            feature = 1
-        else:
-            field_feature_index = field_index
-
-        return "{}:{}:{}".format(field_index, field_feature_index, feature)
-
-    for col_index, col in enumerate(field_names):
-        df_new[col] = df_new[col].apply(lambda x: _convert(col, x, col_index+1, field_feature_dict))
-
-    # Move rating column to the first.
-    field_names.insert(0, col_rating)
-    df_new = df_new[field_names]
-
-    if filepath is not None:
-        np.savetxt(filepath, df_new.values, delimiter=' ', fmt='%s')
-
-    return df_new
-
+    def __init__(self, filepath=None):
+        if filepath is not None:
+            # Check the existence of path
+            import os
+            if not os.path.exists(filepath):
+                raise ValueError("The specified file path {} does not exist".format(self.filepath))
+        
+        self.filepath = filepath
+    
+    def fit(self, df, col_rating=DEFAULT_RATING_COL):
+        """Fit the dataframe for libffm format. In there method does nothing but check the validity of 
+        the input columns
+
+        Args:
+            df (pd.DataFrame): input Pandas dataframe.
+            col_rating (str): rating of the data.
+
+        Return:
+            The instance of the conveter
```

Comment:
typo in converter
Not sure what is the correct naming when we are returning self, could it be:
```
Returns:
   obj: The instance of the converter
```